src/utils/url/getAbsoluteUrl.ts

```diff
@@ -1,9 +1,16 @@
+import { isAbsoluteUrl } from './isAbsoluteUrl'
+
 /**
  * Returns an absolute URL based on the given path.
  */
 export function getAbsoluteUrl(path: string, baseUrl?: string): string {
-  // Ignore absolute URLs.
-  if (!path.startsWith('/')) {
+  // already absolute URL
+  if (isAbsoluteUrl(path)) {
+    return path
+  }
+
+  // Ignore path with pattern start with :*
+  if (/^[:*]/.test(path)) {
```

Comment:
Need to revisit if this change is necessary. Without it, relative paths with patterns will resolve to `:user` -> `http://localhost/:user`, which would then transform to a regular expression. That is the correct behavior.

Reply:
the pattern match effect multiple top-level test case, it's easy to keep the same behavior.
i  agree `:user` should convert to `http://localhost/:user`
```suggestion
  if (/^\*/.test(path)) {
```
`*` as previous discuss, should be left

Reply:
Hm. But `*` won't match `isAbsoluteUrl` and the given path will be returned as-is. Am I missing something?

Reply:
sorry, i mean `*` will catch by regexp and ignored.

I changed the conditions and modified the related unit test cases. 

But code failed in integration test. I see some type declaration errors